browser/tor/onion_location_navigation_throttle.cc

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/tor/onion_location_navigation_throttle.h"
+
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/browser/tor/onion_location_tab_helper.h"
+#include "brave/browser/tor/tor_profile_service.h"
+#include "brave/common/tor/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_window.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+
+namespace tor {
+
+namespace {
+
+bool GetOnionLocation(const net::HttpResponseHeaders* headers,
+                      std::string* onion_location) {
+  onion_location->clear();
+  std::string name = "onion-location";
+
+  if (!headers || !headers->EnumerateHeader(nullptr, name, onion_location))
+    return false;
+  return true;
+}
+
+void OnTorProfileCreated(GURL onion_location,
+                         Profile* profile,
+                         Profile::CreateStatus status) {
+  if (status != Profile::CreateStatus::CREATE_STATUS_INITIALIZED)
+    return;
+  Browser* browser = chrome::FindTabbedBrowser(profile, true);
+  if (!browser)
+    return;
+  content::OpenURLParams open_tor(onion_location, content::Referrer(),
+                                  WindowOpenDisposition::OFF_THE_RECORD,
+                                  ui::PAGE_TRANSITION_TYPED, false);
+  browser->OpenURL(open_tor);
+}
+
+}  // namespace
+
+// static
+std::unique_ptr<OnionLocationNavigationThrottle>
+OnionLocationNavigationThrottle::MaybeCreateThrottleFor(
+    content::NavigationHandle* navigation_handle) {
+  if (tor::TorProfileService::IsTorDisabled())
+    return nullptr;
+  return std::make_unique<OnionLocationNavigationThrottle>(navigation_handle);
+}
+
+OnionLocationNavigationThrottle::OnionLocationNavigationThrottle(
+    content::NavigationHandle* navigation_handle)
+    : content::NavigationThrottle(navigation_handle) {
+  profile_ = Profile::FromBrowserContext(
+      navigation_handle->GetWebContents()->GetBrowserContext());
+}
+
+OnionLocationNavigationThrottle::~OnionLocationNavigationThrottle() {}
+
+content::NavigationThrottle::ThrottleCheckResult
+OnionLocationNavigationThrottle::WillProcessResponse() {
+  if (navigation_handle()->IsInMainFrame()) {
```

Comment:
nit: it's easier to read when 
```
if (!navigation_handle()->IsInMainFrame()) {
  return {};
}
```
is placed at the beginning, and then we go on with the good case

Reply:
actually I think we shouldn't even create the throttle if this is not a main frame navigation. We can check this right in `BraveContentBrowserClient::CreateThrottlesForNavigation`

Reply:
fixed in 08ec907